### PR TITLE
Dave/stalwart 262

### PR DESCRIPTION
**Pull request by daveaugustus**
### :nut_and_bolt: Description: What code changed, and why?
configure errchef service through Automate config

### :+1: Definition of Done
- The configuration needs to be patchable via chef-automate config patch
- The default configuration needs to be set as per the Standalone Chef Server values
- In case of Automate HA, if there is a different default, that needs to be set in the config.toml.erb file
- The Standalone Automate needs to be deployed and tested to see if all the service configurations are in place
- The Automate deployed Chef Server should be running fine
- The  Automate HA needs to be deployed and tested to see if all the service configurations for Chef Server are in place
- The Automate deployed Chef Server in HA should be running fine
- All the changed values need to be documented

### :athletic_shoe: How to Build and Test the Change
```
rebuild components/automate-deployment/
rebuild components/automate-cli/
rebuild components/automate-cs-oc-erchef/
```

- Adde the following conf to a .toml file
```
[erchef.v1.sys.iBrowse]
    ibrowse_max_pipeline_size = 15
    ibrowse_max_sessions = 17
```

and patch using the following command
`chef-automate config patch new_config.toml`

The result should look like this if you run the following command:
```
cat /hab/svc/automate-cs-oc-erchef/config/sys.config
```

Screenshot 2022-11-24 at 2 18 29 PM

### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [x] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### CODE REVIEW

File: components/automate-cs-oc-erchef/habitat/config/sys.config [hunk 1]
```diff
@@ -92,12 +92,14 @@
 
         {authz_timeout, {{cfg.authz.timeout}}},
         {authz_fanout, {{cfg.authz.fanout}}},
+        {authz_cleanup_batch_size, {{cfg.authz.cleanup_batch_size}}},
```
Comment (kalroy) [inline components/automate-cs-oc-erchef/habitat/config/sys.config:95]: We do not need to add a new config. Add the config to L158. 
Modify 
`{cleanup_batch_size, 0}` to `{cleanup_batch_size, {{cfg.authz.cleanup_batch_size}}}`

File: components/automate-cs-oc-erchef/habitat/config/sys.config [hunk 2]
```diff
@@ -92,12 +92,14 @@
...
 
         {reindex_batch_size, {{cfg.index.reindex_batch_size}}},
 
         {enable_actions, false},
         {actions_fqdn, <<"{{cfg.api.actions_fqdn}}">>},
         {ldap, []},
+        {s3_url_ttl, "{{cfg.api.s3_url_ttl}}"},
```
Comment (kalroy) [inline components/automate-cs-oc-erchef/habitat/config/sys.config:102]: We should not add new config. Change in L209
`{s3_url_ttl, 28800},` to `{s3_url_ttl, {{cfg.api.s3_url_ttl}}},`

File: components/automate-cs-oc-erchef/habitat/config/sys.config [hunk 3]
```diff
@@ -160,6 +162,11 @@
         {bulk_fetch_batch_size, {{cfg.api.bulk_fetch_batch_size}}}
     ]},
 
+    {i_browse, [
```
Comment (kalroy) [inline components/automate-cs-oc-erchef/habitat/config/sys.config:165]: Should not add these configs. 
Change this:
`    {ibrowse, [
        {default_max_sessions, 256},
        {default_max_pipeline_size, 1}
    ]},`
    to 
   ```    
    {ibrowse, [
        {default_max_sessions, {{cfg.iBrowse.ibrowse_max_pipeline_size}}},
        {default_max_pipeline_size, {{cfg.iBrowse.ibrowse_max_sessions}}}
    ]}, 
```

**Review by kalroy — CHANGES_REQUESTED** [review]
The changes in sys.config are not correct. Needs to be fixed.

**Comment by dave-augustus** [comment]
Test on automate standalone the result looks fine

```
{cleanup_batch_size, 9}

{s3_url_ttl, 500},

 {ibrowse, [
        {default_max_pipeline_size, 30},
        {default_max_sessions, 300}
    ]},
```

File: api/config/bifrost/config_request.go [hunk 4]
```diff
@@ -36,7 +36,6 @@ func DefaultConfigRequest() *ConfigRequest {
 	c.V1.Sys.Sql.PoolMaxSize = w.Int32(20)
 	c.V1.Sys.Sql.PoolQueueMax = w.Int32(50)
 	c.V1.Sys.Sql.PoolQueueTimeout = w.Int32(2000)
```
Comment (Dmaddu) [inline api/config/bifrost/config_request.go:38]: plz remove the untouched files.

File: api/config/bifrost/config_request.proto [hunk 5]
```diff
@@ -50,7 +50,6 @@ message ConfigRequest {
 		}
```
Comment (Dmaddu) [inline api/config/bifrost/config_request.proto:52]: plz remove the untouched files

**Review by Dmaddu — CHANGES_REQUESTED** [review]
@dave-augustus  Please upload the demo video as well.

**Comment by daveaugustus** [comment]
Automate HA

Screenshot 2022-11-28 at 6 45 30 PM

**Review by kalroy — APPROVED** [review]

**Comment by daveaugustus** [comment]
https://user-images.githubusercontent.com/36985548/204475511-bc785b93-4786-4150-9cea-98dd83d3e932.mov

**Pull request merged**